src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -273,7 +273,7 @@ private function autowireMethod(\ReflectionFunctionAbstract $reflectionMethod, a
                         break;
                     }
 
-                    if (InnerService::class === $attribute->getName()) {
+                    if (Decorated::class === $attribute->getName()) {
                         $arguments[$index] = new Reference($this->currentId.'.inner', ContainerInterface::NULL_ON_INVALID_REFERENCE);
```

Comment:
while at it: in theory, one could change the name of the decorated service reference, so this should be
`$this->container->getDefinition($this->currentId)->getDecoratedService()[1] ?? $this->currentId.'.inner'`
:see_no_evil:

Reply:
`new Reference($decorated[1] ?? $this->currentId.'.inner', $decorated[3] ?? ContainerInterface::NULL_ON_INVALID_REFERENCE)` ? (not sure about the invalid part)

Reply:
looks all good to me

Reply:
for the record, can't work because of https://github.com/symfony/symfony/blob/84d35a2c9adb51b84c81889e351006f89f306049/src/Symfony/Component/DependencyInjection/Compiler/DecoratorServicePass.php#L48